src/ArmTemplates/Extractor/EntityExtractors/APIExtractor.cs

```diff
@@ -250,11 +245,16 @@ public async Task<ApiTemplateResources> GetApiRelatedTemplateResourcesAsync(stri
         /// Generates dictionary with api and api revisions dependency. 
         /// </summary>
         /// <returns></returns>
-        public async Task<Dictionary<string, string>> GenerateApiRevisionDependencyStructure(ExtractorParameters extractorParameters)
+        async Task<Dictionary<string, string>> GenerateApiRevisionsDependenciesAsync(ExtractorParameters extractorParameters)
         {
-            var apis = await this.apisClient.GetAllCurrentAsync(extractorParameters);
             var apiDependency = new Dictionary<string, string>();
+            if (!string.IsNullOrEmpty(extractorParameters.SingleApiName) || !extractorParameters.MultipleApiNames.IsNullOrEmpty())
+            {
+                return apiDependency;
+            }
 
+            var apis = await this.apisClient.GetAllCurrentAsync(extractorParameters);
```

Comment:
lets do `isnullorempty` check before going to `foreach` loop

Reply:
updated